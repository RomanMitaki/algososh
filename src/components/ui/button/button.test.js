import {Button} from "./button";
import renderer from "react-test-renderer";
import {fireEvent, render, screen} from "@testing-library/react";

describe('Тест Button', () => {
    it('Рендер c текстом', () => {
        const button = renderer.create(<Button text='Текст'/>).toJSON();
        expect(button).toMatchSnapshot()
    })

    it('Рендер без текста', () => {
        const button = renderer.create(<Button/>).toJSON();
        expect(button).toMatchSnapshot();
    })

    it('Тест disabled', () => {
        const button = renderer.create(<Button disabled/>).toJSON();
        expect(button).toMatchSnapshot();
    })

    it('Тест лоадера', () => {
        const button = renderer.create(<Button isLoader={true}/>).toJSON();
        expect(button).toMatchSnapshot();
    })

    it('Тест вызова коллбэка при клике на кнопку', () => {
        const mockFn = jest.fn();
        render(<Button text="button" onClick={mockFn}/>);
        const button = screen.getByText('button');
        fireEvent.click(button);
        expect(mockFn).toHaveBeenCalled();
    })

})