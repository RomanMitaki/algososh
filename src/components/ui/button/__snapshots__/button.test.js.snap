// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тест Button Рендер c текстом 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Текст
  </p>
</button>
`;

exports[`Тест Button Рендер без текста 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Тест Button Тест disabled 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Тест Button Тест лоадера 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;
